### PR TITLE
Fix oracle select union sql parse

### PR DESCRIPTION
**Pull request by zihaoAK47**
Fixes #ISSUSE_ID.

Changes proposed in this pull request:
  - The Oracle select SQL statement union was not parsed
  - Fix oracle select union sql parse
  - Add test sql

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ✓] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ✓] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ✓] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ✓] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDMLStatementVisitor.java [hunk 1]
```diff
@@ -669,6 +669,23 @@ public ASTNode visitParenthesisSelectSubquery(final ParenthesisSelectSubqueryCon
...
+            if (null != ctx.UNION(i) && null != ctx.ALL(i)) {
+                combineType = CombineType.UNION_ALL;
+            } else if (null != ctx.UNION(i)) {
+                combineType = CombineType.UNION;
+            } else if (null != ctx.INTERSECT(i)) {
+                combineType = CombineType.INTERSECT;
+            } else {
+                combineType = CombineType.MINUS;
+            }
+            result.setCombine(new CombineSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex(), left,
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDMLStatementVisitor.java:684]: Why call setCombine in foreach? I think combine should be a multi-level nested structure.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]
LGTM, merged.

**Comment by strongduanmu** [comment]
Hi @zihaoAK47, thank you for your contribution. Can you add some more test case for more than one union statement?

**Comment by zihaoAK47** [comment]
> Hi @zihaoAK47, thank you for your contribution. Can you add some more test case for more than one union statement?

Ok, I will add more SQL tests for this parsing

**Pull request merged**